file-check/action.yaml

```
@@ -0,0 +1,54 @@
+name: file-check
+description: ""
+
+inputs:
+  files:
+    description: ""
+    required: true
+  condition:
+    description: ""
+    required: true
+outputs:
+  exists:
+    description: ""
+    value: ${{ steps.file-check.outputs.exists }}
+
+runs:
+  using: composite
+  steps:
+    - name: File check
+      id: file-check
+      run: |
+        set -e
+
+        # Split files into an array
+        input_files="${{ inputs.files }}"
+        file_array=(${input_files})
+
+        # Loop through files
+        exist_files=()
+        for FILE in "${file_array[@]}"; do
+          if [[ -f $FILE ]]; then
+            echo "File $FILE exists"
+            exist_files+=("$FILE")
+          else
+            echo "File $FILE does not exist"
+          fi
+        done
+
+        if [[ ${{ inputs.condition }} = "or" ]]; then
```

I think adding quotes is polite. (Please try checking with `shellcheck`.)

```suggestion
        if [[ "${{ inputs.condition }}" = "or" ]]; then
```